gossip/tx_scrambler_test.go

```diff
@@ -0,0 +1,342 @@
+package gossip
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/exp/rand"
+	"math"
+	"math/big"
+	"reflect"
+	"testing"
+)
+
+func TestTxScrambler_AnalyseEntryList_RemovesDuplicateTransactions(t *testing.T) {
+	entries := createScramblerTestInputRepeatedHashes(2, 10)
```

Comment:
With these test input generation functions being out-factored and parameterized it is very difficult to understand what is actually tested in these functions. 

I would argue that in unit tests clarity is favorable over code re-use. It should be possible for anyone by just reading the code in the test to figure out what the test is about and whether the test itself is doing what the label promises.

A test like this would be much easier to comprehend:
```
func TestTxScrambler_AnalyseEntryList_RemovesDuplicateTransactions2(t *testing.T) {
	entries := []*scramblerEntry{
		{hash: common.Hash{2}},
		{hash: common.Hash{1}},
		{hash: common.Hash{1}},
	}
	entries, _, _ = analyseEntryList(entries)

	want := []*scramblerEntry{
		{hash: common.Hash{2}},
		{hash: common.Hash{1}},
	}
	if !reflect.DeepEqual(want, entries) {
		t.Errorf("want %v, got %v", want, entries)
	}
}
```

Or it might be even extended to multiple different input cases without harming readability a lot:
```
func TestTxScrambler_AnalyseEntryList_RemovesDuplicateTransactions3(t *testing.T) {
	examples := [][]*scramblerEntry{
		{},
		{
			{hash: common.Hash{1}},
		},
		{
			{hash: common.Hash{1}},
			{hash: common.Hash{1}},
		},
		{
			{hash: common.Hash{1}},
			{hash: common.Hash{1}},
			{hash: common.Hash{1}},
		},
		{
			{hash: common.Hash{1}},
			{hash: common.Hash{2}},
			{hash: common.Hash{1}},
		},
		{
			{hash: common.Hash{1}},
			{hash: common.Hash{2}},
			{hash: common.Hash{3}},
			{hash: common.Hash{2}},
			{hash: common.Hash{1}},
		},
	}

	for _, example := range examples {
		t.Run("", func(t *testing.T) {
			result, _, _ := analyseEntryList(example)

			seen := map[common.Hash]struct{}{}
			for _, entry := range result {
				if _, seen := seen[entry.hash]; seen {
					t.Fatalf("duplicate hash %v", entry.hash)
				}
				seen[entry.hash] = struct{}{}
			}
		})
	}
}
```